src/System.Net.Http/src/System/Net/Http/Managed/HttpContentReadStream.cs

```diff
@@ -9,30 +9,14 @@ namespace System.Net.Http
 {
     internal abstract class HttpContentReadStream : HttpContentStream
     {
-        protected HttpConnection _connection;
-
-        public HttpContentReadStream(HttpConnection connection)
+        public HttpContentReadStream(HttpConnection connection) : base(connection)
         {
-            _connection = connection;
         }
 
         public override bool CanRead => true;
-        public override bool CanSeek => false;
         public override bool CanWrite => false;
 
-        public override long Length => throw new NotSupportedException();
-
-        public override long Position
-        {
-            get { throw new NotSupportedException(); }
-            set { throw new NotSupportedException(); }
-        }
-
-        public override void Flush() => throw new NotSupportedException();
-
-        public override long Seek(long offset, SeekOrigin origin) => throw new NotSupportedException();
-
-        public override void SetLength(long value) => throw new NotSupportedException();
+        public override void Flush() { }
```

Comment:
Is this the expected behavior?  Seems like Flush should throw if the stream is not writable.

Reply:
> Is this the expected behavior? Seems like Flush should throw if the stream is not writable.

There's little documentation on what Stream.Flush should do on non-writable streams, and thus our streams are inconsistent, but mostly err on the side of nop'ing.  MemoryStream, FileStream, NetworkStream, and DeflateStream for example all treat Flush on a non-writable stream as a nop.  PipeStream.Flush on a non-writable stream throws.

Reply:
OK